frontend/src/components/EmailConfirmModal/index.tsx

```diff
@@ -0,0 +1,122 @@
+import { useEffect, useState } from 'react';
+import * as S from './EmailConfirmModal.styled';
+import { postVerifyCodeAPi } from 'apis/userApis';
+import useMutation from 'hooks/useMutation';
+import useTimer from 'hooks/useTimer';
+import { User, UserEmailSendRequestBody } from 'types/userType';
+import { getTimestampDiff } from 'utils/timeUtil';
+
+type EmailConfirmModalProps = {
+  email: User['email'];
+  expiredTimestamp: number;
+  onSuccess: () => void;
+  onDismiss: () => void;
+  refetch: (variables: UserEmailSendRequestBody) => Promise<void>;
+};
+
+const EmailConfirmModal: React.FC<
+  React.PropsWithChildren<EmailConfirmModalProps>
+> = ({ email, expiredTimestamp, onSuccess, onDismiss, refetch }) => {
+  const [code, setCode] = useState('');
+  const [isValidCode, setIsValidCode] = useState(false);
+  const [isTimeOver, setIsTimeOver] = useState(false);
+  const { currentTimestamp } = useTimer(0);
+  const remainTime = getTimestampDiff(expiredTimestamp, currentTimestamp);
+
+  const codeVerifyMutation = useMutation(postVerifyCodeAPi, {
+    onSuccess: () => {
+      alert('인증을 완료했습니다.');
+      onSuccess();
+    },
+    onError: () => {
+      alert('인증을 실패했습니다.');
+    },
+  });
+
+  const handleReSendClick = () => {
+    if (confirm('인증 번호를 재요청하시겠습니까?')) {
+      setIsTimeOver(false);
+      refetch({ email });
+    }
+  };
+
+  const handleInputChange: React.ChangeEventHandler<HTMLInputElement> = ({
+    target: { value },
+  }) => {
+    if (/^[0-9]{0,6}$/.test(value)) {
+      setCode(value);
+    }
+  };
+
+  const handleInputKeyDown: React.KeyboardEventHandler<HTMLInputElement> = (
+    event
+  ) => {
+    if (['+', '-', '.', 'E', 'e'].includes(event.key)) {
+      event.preventDefault();
+    }
+  };
+
+  const handleSubmit: React.FormEventHandler<HTMLFormElement> = (event) => {
+    event.preventDefault();
+
+    if (isValidCode && !isTimeOver) {
+      codeVerifyMutation.mutate({
+        email,
+        verifyCode: code,
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (expiredTimestamp - currentTimestamp <= 0) {
+      setCode('');
+      setIsTimeOver(true);
+    }
+  }, [expiredTimestamp, currentTimestamp]);
+
+  useEffect(() => {
+    if (code.length !== 6) {
+      setIsValidCode(false);
+      return;
+    }
+
+    setIsValidCode(true);
+  }, [code]);
+
+  return (
+    <S.Layout>
+      <S.Paragraph>
+        <span>이메일로 전송된</span>
+        <span>인증번호 6자리를 입력해주세요</span>
+      </S.Paragraph>
+      <S.Form onSubmit={handleSubmit}>
+        <S.InputBox>
+          <S.NumberInput
+            type="number"
+            autoComplete="off"
+            placeholder={isTimeOver ? '인증시간 만료' : '123456'}
+            disabled={isTimeOver}
+            name="code"
+            required={true}
+            onChange={handleInputChange}
+            onKeyDown={handleInputKeyDown}
+            value={code}
+          />
```

Comment:
boolean 속성들은 boolean값을 할당하지 않아도 속성명만 할당하면 true로 하기 때문에 다음과 같이 바꿀 수 있을 것 같아요.
[모달](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog#accessibility_considerations)이기 때문에 포커스가능한 요소에 autofocus 주는 것이 좋을 것 같습니다.

```suggestion
          <S.NumberInput
            type="number"
            autoComplete="off"
            placeholder={isTimeOver ? '인증시간 만료' : '123456'}
            disabled={isTimeOver}
            name="code"
            onChange={handleInputChange}
            onKeyDown={handleInputKeyDown}
            value={code}
            required
            autoFocus
          />
```

Reply:
맞습니다. 모달에 autoFocus를 해주는게 사용하기 편리하겠네요!